frida-gum/src/symbol_util.rs

```diff
@@ -0,0 +1,69 @@
+use std::convert::TryInto;
+use std::fmt;
+use std::mem::MaybeUninit;
+use std::{ffi::CStr, str::Utf8Error};
+
+use frida_gum_sys as gum_sys;
+use gum_sys::{gum_symbol_details_from_address, GumDebugSymbolDetails};
+
+use crate::NativePointer;
+
+pub struct DebugSymbolDetails {
+    pub debug_symbol_details: GumDebugSymbolDetails,
+}
+
+impl fmt::Debug for DebugSymbolDetails {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("DebugSymbolDetails")
+            .field("address", &self.address())
+            .field("module_name", &self.module_name())
+            .field("symbol_name", &self.symbol_name())
+            .field("file_name", &self.file_name())
+            .field("line_number", &self.line_number())
+            .finish()
+    }
+}
+
+impl DebugSymbolDetails {
+    pub fn address(&self) -> usize {
+        self.debug_symbol_details.address.try_into().unwrap()
+    }
+
+    pub fn module_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.debug_symbol_details.module_name.as_ptr()) }.to_str()
+    }
+
+    pub fn symbol_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.debug_symbol_details.symbol_name.as_ptr()) }.to_str()
+    }
+
+    pub fn file_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.debug_symbol_details.file_name.as_ptr()) }.to_str()
+    }
+
+    pub fn line_number(&self) -> u32 {
+        self.debug_symbol_details.line_number
+    }
+}
+
+/// Get debug symbol details for address
+///
+/// Look up debug information for address and return it as an object containing:
+/// * address: Address that this symbol is for.
+/// * name: Name of the symbol, as a string.
+/// * module_name: Module name owning this symbol, as a string.
+/// * file_name: File name owning this symbol, as a string.
+/// * line_number: Line number in file_name, as a number.
```

Comment:
This is all redundant for the API of `symbol_details_from_address` these should be docs on each of the respective getters. The only API doc for this function should be:

```
/// Returns debug symbol details for an address.
```

Reply:
Fixed in c4cd2da0db02cac7073ebda3b2cc19d16020c1ea